app/src/main/java/com/noalecohen/dispatcher/api/ApiController.kt

```diff
@@ -0,0 +1,9 @@
+package com.noalecohen.dispatcher.api
+
+import com.noalecohen.dispatcher.api.news.NewsServiceApi
+import com.noalecohen.dispatcher.networking.NetworkManager
+
+open class ApiController {
+    protected var api: NewsServiceApi? = NetworkManager.retrofit?.create(NewsServiceApi::class.java)
```

Comment:
It's great that you made a base class for the ApiContoller!
But, in order to be generic, the class should be abstract and the ApiInterface class should be any class and not NewsServiceApi for example if we will add an API for movies